src/test/java/org/assertj/core/api/object/ObjectAssert_hasSameHashCodeAs_Test.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2017 the original author or authors.
+ */
+package org.assertj.core.api.object;
+
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.api.ObjectAssertBaseTest;
+import org.assertj.core.test.Jedi;
+import org.junit.Test;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for <code>{@link ObjectAssert#hasSameHashCodeAs(Object)}</code>.
+ * 
+ * @author Thomas Traude
+ * @author Rudolf Klassen
+ * @author Matthias Balke
+ */
+public class ObjectAssert_hasSameHashCodeAs_Test extends ObjectAssertBaseTest {
+
+  public static final Jedi OTHER_JEDI = new Jedi("Yoda", "Green");
+
+  @Override
+  protected ObjectAssert<Jedi> invoke_api_method() {
+    return assertions.hasSameHashCodeAs(OTHER_JEDI);
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    verify(objects).assertHasSameHashCodeAs(getInfo(assertions), getActual(assertions), OTHER_JEDI);
+  }
+
+  @Test
```

Comment:
move these tests in a `org.assertj.core.internal.objects.Objects_assertHasSameHashCodeAs_Test`, use `Objects_assertHasToString_Test` as an example (don't hardcode error messages but verify the proper error message factory methods are called).